### ISSUE TITLE
incorrect cursor placement on lines with accented characters

### ISSUE THREAD

**Issue by jtauber** [atom/atom#5975]
There seems to have been a recent regression in the placement of the cursor and end-of-line marker on lines with accented characters. For example 

ὤὤὤὤὤὤὤὤὤὤ

Has ten *characters* but if you ctrl-E to the end of the line the cursor is in the wrong place:

![screen shot 2015-03-15 at 5 18 12 pm](https://cloud.githubusercontent.com/assets/24426/6658154/54d492a6-cb37-11e4-88da-f370ee0b9d59.png)

This may seem obscure but is making it very difficult to use Atom for natural language processing applications with non-ASCII character sets. It worked fairly recently so is likely a recent regression.

**Comment by jtauber** [comment]
Actually, it doesn't even need to involve accented characters.

Entering 

θν

will place the cursor between θ and ν in 0.187.0

**Comment by izuzak** [comment]
@jtauber I'm having trouble reproducing this in 0.187.0. Can you reproduce this in [safe mode](https://atom.io/docs/latest/hacking-atom-debugging#check-if-the-problem-shows-up-in-safe-mode)? Also, which OS are you using and which font?

Here's what I'm seeing:

![cur](https://cloud.githubusercontent.com/assets/38924/6759711/6167f990-cf42-11e4-990b-962e0316c5fd.gif)

**Comment by jtauber** [comment]
OS X 10.10.2 and default font is PT Mono (which doesn't support polytonic Greek so it would be a fall-through font)

**Comment by jtauber** [comment]
I'm going to continue to explore what the reproducible scenario is as if I start a new doc and paste `ὤὤὤὤὤὤὤὤὤὤ` it works fine for me too :/

**Comment by jtauber** [comment]
Until I narrow down the exact scenario, here's a gif of the problem on a real file...

![out](https://cloud.githubusercontent.com/assets/24426/6760980/ae0d5ae2-cf24-11e4-8f97-e956d2311e38.gif)

**Comment by izuzak** [comment]
This also reminds me of https://github.com/atom/atom/issues/1849. Any chance this issue is a duplicate of that?

**Comment by jtauber** [comment]
I suspect the problem can't be reproduced by copy-pasting content here as some Unicode normalization is likely taking place. I have a one-line file that reproduces the problem for me but copy-pasting the content here will likely not work.

**Comment by jtauber** [comment]
It seems entirely plausible this is a duplicate of #1849. If a fix lands for that, I'm happy to validate that it fixes my problem.

**Comment by jtauber** [comment]
In the mean time, let me do a hex dump of the one-line file that exhibits the bug so it's clear exactly what the data on disk is...

```
0000000 ce ac ce b4 ce b9 ce ba ce ad cf 89 2c 20 ce ac
0000010 ce b4 ce b9 ce ba ce ae cf 83 cf 89 2c 20 e1 bc
0000020 a0 ce b4 ce af ce ba ce b7 cf 83 ce b1 2c 20 e1
0000030 bc a0 ce b4 ce af ce ba ce b7 ce ba ce b1 2c 20
0000040 e1 bc a0 ce b4 ce af ce ba ce b7 ce bc ce b1 ce
0000050 b9 2c 20 e1 bc a0 ce b4 ce b9 ce ba ce ae ce b8
0000060 ce b7 ce bd 0a                                 
0000065
```

**Comment by izuzak** [comment]
@jtauber Since they are indeed similar, what do you think about adding a comment to that issue with a description of what you're seeing (and a link to the file which reproduces this for you), closing this issue and waiting for a fix for that issue to land? Once it does, we can reopen this if it's not fixed as well.

**Comment by jtauber** [comment]
@izuzak sure, I'm happy to do that.

**Comment by jtauber** [comment]
This is probably the best example I can come up with of the bug and it's effect:

1) the existence of the macron on the alphas makes every glyph on the line narrower (including all those earlier in the line)
2) the *cursor* placement ignores this narrowing and the cursor's x offset on the second line is where it would be on the first line

![screen shot 2015-06-06 at 2 16 12 am](https://cloud.githubusercontent.com/assets/24426/8020407/72580e3a-0c45-11e5-8ab9-96c171c5b159.png)

**Comment by jtauber** [comment]
Note, the reason I'm keeping this ticket updated, even though closed, is because it's only tentatively closed until #1849 is fixed to see if it's a related issue. Because my issue is more specific, I want to make sure all the information is available here in case fixing #1849 doesn't not fix the specific issue described here.

**Comment by jtauber** [comment]
I've reopened this as it really is a separate issue from #1849 and can be fixed independently.

I will verify this is fixed once #6055 and #6083 land.

**Comment by benogle** [comment]
Closing as the new text measurement bits in #6083 are in master

**Comment by jtauber** [comment]
I can verify the problems described here are all resolved in 1.1
